### DISCUSSION TITLE
Make QueryBuilder::$request accessible

### DISCUSSION THREAD

**Discussion by callmez** [spatie/laravel-query-builder#674] [Ideas]
Hi @AlexVanderbist, @freekmurze,  Currently QueryBuilder::$request can't be access outside and there is no way to pass custom QueryBuilderRequest to QueryBuilder.
 
 it would be nice if add a request getter. thanks!

**Comment by AlexVanderbist** [comment]
Hey @callmez,

If I recall, you can use `QueryBuilder::for(User::class, $request)` where `$request` is a custom `QueryBuilderRequest`.

Regarding a getter for the `QueryBuilderRequest`: feel free to send a PR to add that method and I'll gladly merge it :)
